agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/AlleleRepository.java

```diff
@@ -75,11 +102,7 @@ public List<String> getAllAlleleKeys() {
 		return list;
 	}
 
-	private List<String> allAlleleIDs = null;
-
 	public List<String> getAllAlleleIDs() {
-		if (allAlleleIDs != null)
-			return null;
```

Comment:
@cmpich the logic here should have been: `if (allAlleleIDs != null) return allAlleleIDs;`

Reply:
Plus there is no need to cache these as they should only get called a few times.

Reply:
But I don't think this causes too much memory strain on the indexer. This will increase the timing of the indexer.